core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java

```diff
@@ -144,18 +149,26 @@ public void execute() throws CommandExecutionException {
             TestPlan testPlan = generateTestPlan(deploymentPattern, testConfig, scenarioRepoDir, infraRepo);
 
             //Set the log file path
-            LogFilePathLookup.setLogFilePath(deriveLogFilePath(product, testPlan));
+            LogFilePathLookup.setLogFilePath(deriveLogFilePath(testPlan));
 
             // Product, deployment pattern, test plan and test scenarios should be persisted
             testPlan = persistTestPlan(testPlan);
 
             // Execute test plan
-            executeTestPlan(testPlan, infrastructure);
+            testPlan = executeTestPlan(testPlan, infrastructure);
+
+            // Generate report for the test plan
+            TestReportEngine testReportEngine = new TestReportEngine();
```

Comment:
Shall we use dependency injection (DI) here to get an instance of TestReportEngine? We shouldn't create instances unnecessarily like this nor use singletons since we have a better alternative. 

See Spring DI: http://www.vogella.com/tutorials/SpringDependencyInjection/article.html#dependencyinjection

Reply:
Created a git issue to track this - https://github.com/wso2-incubator/testgrid/issues/322